server/test/unit/server/webservices/views/test_permissions.py

```diff
@@ -0,0 +1,186 @@
+
+        mock_resp.assert_called_once_with(expected_cont)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.permissions.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.permissions.factory')
+    def test_get_resource_permission(self, mock_f, mock_resp):
+        """
+        Test specific resource permissions retrieval.
+        """
+        perm = {'resource': '/v2/some/', 'id': '1234',
+                'users': [{'username': 'test-user', 'permissions': [0]}]}
+
+        mock_f.permission_query_manager.return_value.find_by_resource.return_value = perm
+        mock_f.permission_manager.return_value.operation_value_to_name.return_value = 'READ'
+
+        request = mock.MagicMock()
+        request.GET = {'resource': '/v2/some/'}
+
+        permission = PermissionView()
+        response = permission.get(request)
+        expected_cont = [{'id': '1234', 'resource': '/v2/some/', 'users': {'test-user': ['READ']}}]
+
+        mock_resp.assert_called_once_with(expected_cont)
+        self.assertTrue(response is mock_resp.return_value)
+
+
+class TestGrantToUserView(unittest.TestCase):
+    """
+    Test grant permission to user.
+    """
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_EXECUTE())
+    @mock.patch('pulp.server.webservices.views.permissions.generate_json_response')
+    @mock.patch('pulp.server.webservices.views.permissions.factory')
+    def test_grant_to_user(self, mock_factory, mock_resp):
+        """
+        Test grant permissions to user.
+        """
+        request = mock.MagicMock()
+        request.body_as_json = {'operations': ['READ'], 'login': 'test', 'resource': '/v2/some/'}
+        mock_factory.permission_manager.return_value.grant.return_value = None
+        mock_factory.permission_manager.return_value.operation_names_to_values.return_value = [0]
+        grant = GrantToUserView()
+        response = grant.post(request)
+
+        mock_resp.assert_called_once_with(None)
+        self.assertTrue(response is mock_resp.return_value)
+        mock_factory.permission_manager.return_value.grant.assert_called_once_with(
+            '/v2/some/', 'test', [0])
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_EXECUTE())
+    def test_grant_to_user_invalid_param(self):
+        """
+        Test grant permissions to user with invalid params.
+        """
+        request = mock.MagicMock()
+        request.body_as_json = {'operations': ['READ'], 'resource': '/v2/some/'}
+        grant = GrantToUserView()
+        try:
+            grant.post(request)
+        except InvalidValue, response:
```

Comment:
This may be problematic because you won't have a reference to the 'response' object anymore. Perhaps that is why it is written this way. It's up to you how best to fix it (if any).